lib/activerecord-import/import.rb

```diff
@@ -230,8 +230,49 @@ def supports_on_duplicate_key_update?
     # This returns an object which responds to +failed_instances+ and +num_inserts+.
     # * failed_instances - an array of objects that fails validation and were not committed to the database. An empty array if no validation is performed.
     # * num_inserts - the number of insert statements it took to import the data
-    def import( *args )
-      options = { :validate=>true, :timestamps=>true }
+    def import(*args)
+      if args.first.is_a?( Array ) and args.first.first.is_a? ActiveRecord::Base
+        options = {}
+        options.merge!( args.pop ) if args.last.is_a?(Hash)
+
+        models = args.first # the import argument parsing is too tangled for me ... I only want to prove the concept of saving recursively
+        result = import_helper(models, options)
+        # now, for all the dirty associations, collect them into a new set of models, then recurse.
+        # notes:
+        #    does not handle associations that reference themselves
+        #    assumes that the only associations to be saved are marked with :autosave
+        #    should probably take a hash to associations to follow.
+        hash={}
+        models.each {|model| add_objects(hash, model) }
+
+        hash.each_pair do |class_name, assocs|
+          assocs.each_pair do |assoc_name, subobjects|
+            subobjects.first.class.import(subobjects, options) unless subobjects.empty?
+          end
+        end
+        result
+      else
+        import_helper(*args)
+      end
+    end
+
+    def add_objects(hash, parent)
```

Comment:
We'll want to hide this method and import_helper in the Import module. Right now it looks like it's being added to ActiveRecord::Base as a class method. We should only add the public methods that we want to expose to a user there so we don't unnecessarily pollute or add potential conflicts w/ other things on ActiveRecord::Base.